### PR TITLE
feat(core): Save Sub-Workflow and Parent-Workflow Execution IDs

### PR DESCRIPTION
**Pull request by janober**
## Summary
Saves the Execution ID of sub and parent workflows and so makes it now easily possible to jump to the corresponding sub or parent workflow.

In Parent-Workflow:
Screenshot from 2024-02-11 15-24-08
Screenshot from 2024-02-11 15-24-24

In Sub-Workflow:
Screenshot from 2024-02-11 15-24-45

In AI Logs:
Screenshot from 2024-03-08 16-16-57

Is currently still incomplete:
- Spend no time thinking about UX/UI 
- Tests are missing (created backend-node tests but realized that they were never fully implemented, so we probably have to create frontend-tests instead)
- Linking with Error Workflows
- Have to implement what happens when a sub-workflow is supplied not via Workflow-ID. Realized that it is currently totally broke so will look into that once that functionality works again.

## Review / Merge checklist
- [x] PR title and summary are descriptive. **Remember, the title automatically goes into the changelog. Use `(no-changelog)` otherwise.** ([conventions](https://github.com/n8n-io/n8n/blob/master/.github/pull_request_title_conventions.md))
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included.
   > A bug is not considered fixed, unless a test is added to prevent it from happening again.
   > A feature is not complete without tests.

### CODE REVIEW

**Comment by jesper-bylund** [comment]
@janober this is a great start. As an MVP UX is ok, I would change the phrases to be more descriptive: 
"Open Execution 44916" -> "Open sub workflow execution 44916"
"Select execution" -> "Select sub workflow execution"
"Open Execution 44913" -> "Open parent execution 44913"

File: packages/editor-ui/src/plugins/i18n/locales/en.json [hunk 1]
```diff
@@ -830,6 +830,9 @@
 	"ndv.output.noOutputData.title": "No output data returned",
 	"ndv.output.noOutputDataInBranch": "No output data in this branch",
 	"ndv.output.of": " of ",
+	"ndv.output.subworkflow.openExecution": "Open Execution \"{executionId}\"",
```
Comment (jesper-bylund) [inline packages/editor-ui/src/plugins/i18n/locales/en.json:833]:
```suggestion
	"ndv.output.subworkflow.openExecution": "Open sub workflow execution \"{executionId}\"",
```

File: packages/editor-ui/src/plugins/i18n/locales/en.json [hunk 2]
```diff
@@ -830,6 +830,9 @@
...
+	"ndv.output.subworkflow.selectExecution": "Select Execution",
```
Comment (jesper-bylund) [inline packages/editor-ui/src/plugins/i18n/locales/en.json:834]:
```suggestion
	"ndv.output.subworkflow.selectExecution": "Select sub workflow execution",
```

File: packages/editor-ui/src/plugins/i18n/locales/en.json (hunk 1 above)
Comment (jesper-bylund) [inline packages/editor-ui/src/plugins/i18n/locales/en.json:833]:
```suggestion
	"ndv.output.subworkflow.openExecution": "Open Execution \"{executionId}\"",
	"ndv.output.parentworkflow.openExecution": "Open parent workflow execution \"{executionId}\"",
```

**Comment by janober** [comment]
Thanks a lot @jesper-bylund and also thanks for making it easy to apply.

**Comment by mutdmour** [comment]
@janober working on bringing this feature to life, starting with this branch. 

What's left here that you did not get to chance to finish? Or what do you think I need to pay attention to?

**Comment by janober** [comment]
Very good question. It was quite a while ago. Generally, think what is written in the PR description is mainly it:
- Improve UI/UX
- Create tests, test workflow did not work as executing sub-workflows is something that did not get implemented. So either that would have to get implemented (generally for sure helpful, but also do not know how much work it would be) or at least frontend tests, as there it should be simple.
- What is kind of optional, is to also link up Error Workflows. So that it is also possible to jump from a failed execution to the error workflow execution or the other way around. But that is nice to have and could also be added later.
- When a sub-workflow gets set on the Execute Workflow node not via ID but rather, for example, via JSON, file or URL. If I remember correctly did I test in the end with JSON and it worked. Think never ended up testing with file or URL.

**Comment by mutdmour** [comment]
closing in favor of https://github.com/n8n-io/n8n/pull/11602

**Pull request closed without merging**